pkg/controllers/subnet/subnet_poll.go

```diff
@@ -0,0 +1,254 @@
+		return
+	}
+}
+
+// pollSingleSharedSubnetWithError updates a subnet CR with error information
+// This is used when either GetNSXSubnetByAssociatedResource or GetSubnetStatus returns an error
+func (r *SubnetReconciler) pollSingleSharedSubnetWithError(ctx context.Context, namespacedName types.NamespacedName, err error, associatedResource string, errorType string) {
+	// Get the subnet CR
+	subnetCR := &v1alpha1.Subnet{}
+	if getErr := r.Client.Get(ctx, namespacedName, subnetCR); getErr != nil {
+		r.handleSubnetGetError(getErr, namespacedName)
+		return
+	}
+
+	// Skip if the subnet is being deleted
+	if !subnetCR.DeletionTimestamp.IsZero() {
+		r.removeSubnetFromPollingQueue(namespacedName, "deleting")
+		return
+	}
+
+	// Set the subnet ready status to false with the appropriate error message
+	errorMsg := fmt.Sprintf("Failed to get %s during polling for AssociatedResource %s: %v", errorType, associatedResource, err)
+	r.clearSubnetAddresses(subnetCR)
+	setSubnetReadyStatusFalse(r.Client, ctx, subnetCR, metav1.Now(), err, errorMsg)
+	log.Info(fmt.Sprintf("Set subnet ready status to false due to %s failure", errorType), "Subnet", namespacedName)
+}
+
+// handleSubnetGetError handles errors when getting a subnet CR
+func (r *SubnetReconciler) handleSubnetGetError(err error, namespacedName types.NamespacedName) {
+	if apierrors.IsNotFound(err) {
+		// Subnet CR no longer exists, remove it from the polling queue
+		r.removeSubnetFromPollingQueue(namespacedName, "deleted")
+	} else {
+		log.Error(err, "Failed to get Subnet CR during polling", "Subnet", namespacedName)
+	}
+}
+
+// removeSubnetFromPollingQueue removes a subnet from the polling queue
+func (r *SubnetReconciler) removeSubnetFromPollingQueue(namespacedName types.NamespacedName, reason string) {
+	r.sharedSubnetsMutex.Lock()
+	defer r.sharedSubnetsMutex.Unlock()
+	delete(r.sharedSubnetsMap, namespacedName)
+	log.Info(fmt.Sprintf("Removed %s subnet from polling queue", reason), "Subnet", namespacedName)
+}
+
+// updateSubnetIfNeeded updates the subnet if it has changed
+func (r *SubnetReconciler) updateSubnetIfNeeded(ctx context.Context, subnetCR *v1alpha1.Subnet, nsxSubnet *model.VpcSubnet,
+	statusList []model.VpcSubnetStatus, namespacedName types.NamespacedName) error {
+	// Create a copy of the current status and spec before updating
+	originalStatus := subnetCR.Status.DeepCopy()
+	originalSpec := subnetCR.Spec.DeepCopy()
+
+	r.SubnetService.MapNSXSubnetToSubnetCR(subnetCR, nsxSubnet)
+	r.SubnetService.MapNSXSubnetStatusToSubnetCRStatus(subnetCR, statusList)
+
+	// Check if spec has changed
+	specChanged := r.hasSubnetSpecChanged(originalSpec, &subnetCR.Spec)
+	// Check if the status has changed
+	statusChanged := r.hasStatusChanged(originalStatus, &subnetCR.Status)
+
+	// Update the CR if either spec or status has changed
+	if specChanged {
+		// Update the CR spec if there are changes
+		if err := r.Client.Update(ctx, subnetCR); err != nil {
```

Comment:
Have you tested this case? I'm not confident about if it works as Update might only updates spec and metatdata and left status unchanged.
But unfortunately I didn't find a formal doc regarding this. This most close one I find is here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status It mentions
```
For example, the /pods endpoint only allows updates to metadata and spec, since those reflect end-user intent. An automated process should be able to modify status for users to see by sending an updated Pod kind to the server to the "/pods/<name>/status" endpoint 
```

Reply:
for safety, remove else